### PR TITLE
Makes the New Guns Acquirable via RND + Reformats Gun Naming Scheme

### PR DESCRIPTION
**Pull request by genessee-forgot-his-password-again**
## Description of changes
Does a (hopefully) functional version of what I tried to do in #355 - see there for the actual changes listed out.

In addition to those changes, this PR also improves the gun names and descriptions to provide more useful information such as caliber and a brief description of the guns' weaknesses.

## Why and what will this PR improve
Makes the assortment of new guns obtainable in-game.

Also makes legacy guns uncraftable, as they cannot be removed without fooling with stuff outside the modpack.

Having this new naming format for guns makes them more immediately useful to people who don't know, for example, what caliber the weapon needs. Also makes them less of a text-wall trap to examiners.

## Authorship
genessee made this

## Changelog
:cl:
add: An assortment of guns are now craftable via RND
tweak: The Footman rifle is now huge instead of bulky
tweak: The Midas pistol now requires a lower materials tech to craft
tweak: All firearms now have a more concise naming and description format
fix: The legacy guns from Nebula are no longer craftable via RND
/:cl:

### CODE REVIEW

**Comment by genessee-forgot-his-password-again** [comment]
ah shit i didn't do it in github desktop which is why my thing got denied in the first place

oooooops!!!!

**Comment by genessee-forgot-his-password-again** [comment]
okay i think i fixed it

**Comment by genessee-forgot-his-password-again** [comment]
not really sure why map tests are failing

**Comment by genessee-forgot-his-password-again** [comment]
updated the naming format

**Comment by genessee-forgot-his-password-again** [comment]
oh hey look i fixed it

File: mods/persistence/modules/fabrication/fabricator_protolathe.dm [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/obj/machinery/fabricator/protolathe
+	base_storage_capacity = list(
```
Comment (NataKilar) [inline mods/persistence/modules/fabrication/fabricator_protolathe.dm]: This change should no longer be necessary (fabricators build their material capacity from their design list)

File: mods/persistence/modules/projectiles/overrides.dm [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/obj/item/gun/launcher/syringe/rapid
+	origin_tech = "{'wormholes':1337}"
+
+/obj/item/gun/energy/temperature
```
Comment (NataKilar) [inline mods/persistence/modules/projectiles/overrides.dm]: I don't remember this gun being particularly good, and it doesn't have a new analogue. I don't think it needs to be made unobtainable. Ditto for a few others below

File: mods/persistence/modules/projectiles/overrides.dm [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	origin_tech = "{'wormholes':1337}"
+
+/obj/item/grenade/anti_photon
```
Comment (NataKilar) [inline mods/persistence/modules/projectiles/overrides.dm]: Ditto

File: mods/persistence/modules/projectiles/overrides.dm [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	origin_tech = "{'wormholes':1337}"
+
+/obj/item/gun/energy/confuseray
```
Comment (NataKilar) [inline mods/persistence/modules/projectiles/overrides.dm]: Ditto

File: mods/persistence/modules/projectiles/overrides.dm [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+/obj/item/gun/energy/xray
+	origin_tech = "{'wormholes':1337}"
+
+/obj/item/gun/launcher/grenade
+	origin_tech = "{'wormholes':1337}"
+
+/obj/item/gun/magnetic/railgun/flechette
+	origin_tech = "{'wormholes':1337}"
+
+/obj/item/gun/energy/toxgun
```
Comment (NataKilar) [inline mods/persistence/modules/projectiles/overrides.dm]: Ditto

File: mods/persistence/modules/projectiles/overrides.dm [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+	origin_tech = "{'wormholes':1337}"
+
+/obj/item/gun/energy/decloner
```
Comment (NataKilar) [inline mods/persistence/modules/projectiles/overrides.dm]: Ditto? I have no idea what this one does.

File: mods/persistence/modules/projectiles/overrides.dm [hunk 7]
```diff
@@ -0,0 +1,38 @@
+/obj/item/gun/launcher/syringe/rapid
+	origin_tech = "{'wormholes':1337}"
```
Comment (NataKilar) [inline mods/persistence/modules/projectiles/overrides.dm]: Setting origin tech like this should be unnecessary. Instead set ``research_excluded = TRUE``

**Review by NataKilar — CHANGES_REQUESTED** [review]

File: mods/persistence/modules/projectiles/overrides.dm (hunk 6 above)
Comment (genessee-forgot-his-password-again) [reply mods/persistence/modules/projectiles/overrides.dm]: Deals cellular damage

File: mods/persistence/modules/projectiles/overrides.dm (hunk 7 above)
Comment (genessee-forgot-his-password-again) [reply mods/persistence/modules/projectiles/overrides.dm]: done

File: mods/persistence/modules/fabrication/fabricator_protolathe.dm (hunk 1 above)
Comment (genessee-forgot-his-password-again) [reply mods/persistence/modules/fabrication/fabricator_protolathe.dm]: unecessary file deleted

File: mods/persistence/modules/projectiles/overrides.dm (hunk 2 above)
Comment (genessee-forgot-his-password-again) [reply mods/persistence/modules/projectiles/overrides.dm]: as discussed on discord, will add variants for these guns in another pr soon

File: mods/persistence/modules/projectiles/overrides.dm (hunk 3 above)
Comment (genessee-forgot-his-password-again) [reply mods/persistence/modules/projectiles/overrides.dm]: didnt realize this was a grenade lol 
removed from override

File: mods/persistence/modules/projectiles/overrides.dm (hunk 4 above)
Comment (genessee-forgot-his-password-again) [reply mods/persistence/modules/projectiles/overrides.dm]: as discussed on discord, will add variants for these guns in another pr soon

File: mods/persistence/modules/projectiles/overrides.dm (hunk 5 above)
Comment (genessee-forgot-his-password-again) [reply mods/persistence/modules/projectiles/overrides.dm]: as discussed on discord, will add variants for these guns in another pr soon

File: mods/persistence/modules/projectiles/overrides.dm (hunk 6 above)
Comment (genessee-forgot-his-password-again) [reply mods/persistence/modules/projectiles/overrides.dm]: as discussed on discord, will add variants for these guns in another pr soon

**Review by NataKilar — APPROVED** [review]

**Pull request merged**